packages/functions/src/dal/AdminDataAccessLayer.ts

```diff
@@ -60,3 +65,37 @@ export async function getCoordinatorDonors(
 
   return donors;
 }
+
+export function getValidHospitalsOrThrow(
+  coordinator: DbCoordinator,
+  hospital: Hospital | typeof HospitalUtils.ALL_HOSPITALS_SELECT
+): Hospital[] {
+  const allActiveHospitalsFiltered =
+    hospital === HospitalUtils.ALL_HOSPITALS_SELECT
+      ? HospitalUtils.activeHospitals
+      : [hospital];
+
+  switch (coordinator.role) {
+    case CoordinatorRole.SYSTEM_USER:
+      return allActiveHospitalsFiltered;
+    case CoordinatorRole.ZM_COORDINATOR:
+      return allActiveHospitalsFiltered;
+    case CoordinatorRole.HOSPITAL_COORDINATOR:
+      if (hospital === HospitalUtils.ALL_HOSPITALS_SELECT) {
+        return coordinator.hospitals ?? [];
+      }
+      if (!coordinator.hospitals?.includes(hospital)) {
+        console.error(
+          `Coordinator ${coordinator.id} ${coordinator.role} ${coordinator.hospitals} missing permissions for ${hospital}`
+        );
+        throw Error("Coordinator has no permissions for hospital");
+      }
+      return [hospital];
+    case CoordinatorRole.GROUP_COORDINATOR:
+      return hospital === HospitalUtils.ALL_HOSPITALS_SELECT
+        ? coordinator.hospitals ?? []
+        : [hospital];
+  }
+
+  throw Error("Unfamiliar coordinator role");
```

Comment:
Unfamiliar coordinator hospital

Reply:
if the code gets there it means that we have a role that was not covered in the switch case